app/services/editions/db/CollectionsQueries.scala

```diff
@@ -93,17 +50,67 @@ trait CollectionsQueries {
     """.execute().apply()
 
     collection.items.zipWithIndex.foreach { case (article, index) =>
+      val addedOn = EditionsDB.dateTimeFromMillis(article.addedOn)
       sql"""
           INSERT INTO articles (
           collection_id,
           page_code,
           index,
           added_on,
           metadata
-          ) VALUES (${collection.id}, ${article.pageCode}, $index, now(), ${article.metadata.map(m => Json.toJson(m).toString)}::JSONB)
+          ) VALUES (${collection.id}, ${article.pageCode}, $index, $addedOn, ${article.metadata.map(m => Json.toJson(m).toString)}::JSONB)
        """.execute().apply()
     }
 
-    collection
+    val rows = fetchCollectionsSql(where = sqls"collections.id = ${collection.id}").apply()
+
+    val updatedCollections = convertRowsToCollections(rows)
+
+    // we have filtered on a single id so this list should only contain one collection
+    assert(updatedCollections.size == 1, s"Retrieved ${updatedCollections.size} collections from DB but there should be exactly one. Failing fast.")
```

Comment:
would pushing this into `convertRowsToCollections` be a cleaner pattern?

Reply:
`convertRowsToCollections` can legitimately return multiple collections (see other call site of the function).